testutil/integration/suites/base.go

```diff
@@ -0,0 +1,93 @@
+//go:build integration
+
+package suites
+
+import (
+	"testing"
+
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/pokt-network/poktroll/app/volatile"
+	"github.com/pokt-network/poktroll/testutil/integration"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	gatewaytypes "github.com/pokt-network/poktroll/x/gateway/types"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+	tokenomicstypes "github.com/pokt-network/poktroll/x/tokenomics/types"
+)
+
+// TODO_IMPROVE: Ideally this list should be populated during integration app construction.
+var allPoktrollModuleNames = []string{
+	sharedtypes.ModuleName,
+	sessiontypes.ModuleName,
+	servicetypes.ModuleName,
+	apptypes.ModuleName,
+	gatewaytypes.ModuleName,
+	suppliertypes.ModuleName,
+	prooftypes.ModuleName,
+	tokenomicstypes.ModuleName,
+}
+
+var _ IntegrationSuite = (*BaseIntegrationSuite)(nil)
+
+// BaseIntegrationSuite is a base implementation of IntegrationSuite.
+// It is intended to be embedded in other integration test suites.
+type BaseIntegrationSuite struct {
+	suite.Suite
+	app *integration.App
+}
+
+// NewApp constructs a new integration app and sets it on the suite.
+func (s *BaseIntegrationSuite) NewApp(t *testing.T) *integration.App {
+	t.Helper()
+
+	s.app = integration.NewCompleteIntegrationApp(t)
+	return s.app
+}
+
+// SetApp sets the integration app on the suite.
+func (s *BaseIntegrationSuite) SetApp(app *integration.App) {
```

Comment:
Do we need this given that it's already initialized in `NewApp`?

Reply:
I think it's prudent. I can imagine a scenario where the integration app is constructed elsewhere. In the case where we want a test suite to inherit from multiple other suites, we would require this method because only one suite can be embedded and the other(s) would still need a way to set their app fields to the embedded app (rather than constructing new one(s)).